### PR TITLE
courses and ecosystems metadata internal

### PR DESCRIPTION
**Pull request by pandafulmanda**

### CODE REVIEW

File: lib/active_record_extension.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+module ActiveRecordExtension
+
+  extend ActiveSupport::Concern
+
+  module ClassMethods
+    def pluck_with_keys(*keys)
+      self.pluck(*keys).map{ |plucked_values|
+        Hash[keys.zip(plucked_values.kind_of?(Array)? plucked_values : [plucked_values])]
```
Comment (Dantemss) [inline lib/active_record_extension.rb]: I would usually do [plucked_values].flatten but no idea if that's good/bad performance-wise...

File: app/controllers/courses_controller.rb [hunk 2]
```diff
@@ -53,4 +63,33 @@ def _create_response_payload_schema
...
+        },
+      },
+      'required': ['course_responses'],
+      'additionalProperties': false,
+      'standard_definitions': _standard_definitions,
+      'definitions': {
+        'course_metadata': {
+          'type': 'object',
+          'properties': {
+            'uuid': {'$ref': '#standard_definitions/uuid'}
```
Comment (Dantemss) [inline app/controllers/courses_controller.rb]: Might wanna return the ecosystem uuid here (maybe call it `initial_ecosystem_uuid` to make it clear that we won't bother trying to find the latest ecosystem).

The reason for that is that we might want to filter what courses we care about based on the ecosystem.

Another useful key is `is_demo` but that would require changing the create_course schema to add that information.

File: app/controllers/ecosystems_controller.rb [hunk 3]
```diff
@@ -128,4 +139,33 @@ def _create_response_payload_schema
...
+        },
+      },
+      'required': ['ecosystem_responses'],
+      'additionalProperties': false,
+      'standard_definitions': _standard_definitions,
+      'definitions': {
+        'ecosystem_metadata': {
+          'type': 'object',
+          'properties': {
+            'uuid': {'$ref': '#standard_definitions/uuid'}
```
Comment (Dantemss) [inline app/controllers/ecosystems_controller.rb]: Might wanna ask Kevin if he plans to filter ecosystems on UUID or if we need something like a book title/subject.

File: lib/active_record_extension.rb (hunk 1 above)
Comment (pandafulmanda) [reply lib/active_record_extension.rb]: hmmm :thinking:

**Pull request merged**